packages/devtools_app/lib/src/ui/analytics_platform.dart

```diff
@@ -17,21 +17,20 @@ import '../ui/analytics.dart' as ga;
 @JS('getDevToolsPropertyID')
 external String devToolsProperty();
 
-@JS('gaStorageCollect')
-external String storageCollectValue();
+@JS('initializeGA')
+external void initializeGA();
 
-@JS('gaStorageDontCollect')
-external String storageDontCollectValue();
+@JS('hookupListenerForGA')
+external void jsHookupListenerForGA();
 
-bool isAnalyticsAllowed() =>
-    html.window.localStorage[devToolsProperty()] == storageCollectValue();
+Future<bool> get isAnalyticsAllowed async => await ga.isEnabled;
 
-void setAllowAnalytics() {
-  html.window.localStorage[devToolsProperty()] = storageCollectValue();
+void setAllowAnalytics() async {
+  await ga.setEnabled(true);
 }
 
-void setDontAllowAnalytics() {
-  html.window.localStorage[devToolsProperty()] = storageDontCollectValue();
+void setDontAllowAnalytics() async {
+  await ga.setEnabled(false);
```

Comment:
why not combine these into `setAllowAnalytics(bool allow)`

Reply:
I thought it was a good idea to explicitly show analytics being collected or not.  But could have been done as you suggested too.